qiskit/dagcircuit/collect_blocks.py

```diff
@@ -86,22 +87,48 @@ def _op_nodes(self):
             return self.dag.get_nodes()
 
     def _direct_preds(self, node):
-        """Returns direct predecessors of a node."""
+        """Returns direct predecessors of a node. This function takes into account the
+        direction of collecting blocks, that is node's predecessors when collecting
+        backwards are the direct successors of a node in the DAG.
+        """
         if not self.is_dag_dependency:
-            return [pred for pred in self.dag.predecessors(node) if isinstance(pred, DAGOpNode)]
+            if self._collect_from_back:
+                return [pred for pred in self.dag.successors(node) if isinstance(pred, DAGOpNode)]
+            else:
+                return [pred for pred in self.dag.predecessors(node) if isinstance(pred, DAGOpNode)]
```

Comment:
I feel like this is a missing part of the dagcircuit api to be able to do this in a single method call. But we can look at doing that in a follow up.

Reply:
I agree that having functions like `dag.op_predecessors` or `dag.op_successors` would be useful.